kolibri/core/content/tasks_utils.py

```diff
@@ -0,0 +1,98 @@
+from rest_framework import serializers
+
+from kolibri.core.content.models import ChannelMetadata
+from kolibri.core.content.utils.channels import get_mounted_drive_by_id
+from kolibri.core.discovery.models import NetworkLocation
+from kolibri.utils import conf
+
+
+def add_drive_info(import_task, task_description):
+    try:
+        drive_id = task_description["drive_id"]
+    except KeyError:
+        raise serializers.ValidationError("The drive_id field is required.")
+
+    try:
+        drive = get_mounted_drive_by_id(drive_id)
+    except KeyError:
+        raise serializers.ValidationError(
+            "That drive_id was not found in the list of drives."
+        )
+
+    import_task.update({"drive_id": drive_id, "datafolder": drive.datafolder})
+
+    return import_task
+
+
+def get_channel_name(channel_id, require_channel=False):
+    try:
+        channel = ChannelMetadata.objects.get(id=channel_id)
+    except ChannelMetadata.DoesNotExist:
+        channel = None
+        if require_channel:
+            raise serializers.ValidationError("This channel does not exist.")
+
+    return "" if channel is None else channel.name
+
+
+def validate_content_task(request, task_description, require_channel=False):
+    try:
+        channel_id = task_description["channel_id"]
+    except KeyError:
+        raise serializers.ValidationError("The channel_id field is required.")
+
+    try:
+        channel_name = task_description["channel_name"]
+    except KeyError:
+        channel_name = get_channel_name(
+            channel_id=channel_id, require_channel=require_channel
+        )
```

Comment:
Earlier it was: 

```python
channel_name = task_description.get("channel_name", 
               get_channe_name(channel_id=channel_id, require_channel=require_channel))
```

The above was interfering with tests. Even when `channel_name` was provided, `get_channel_name` was being run so I changed it to `try except`.

Is the new code segment doing the right thing?
If we have `channel_id` already shouldn't we fetch from db and return that as `channel_name`...? This will return the correct channel name even if our frontend provides errenous `channel_name`. 

Thoughts on these @rtibbles?

Reply:
The main issue is for channels that do not yet exist in our database, so we can't look up the channel name (which is why it need to be provided separately).

Reply:
Gotcha!